pygmt/tests/test_clib_virtualfile_in.py

```diff
@@ -101,3 +103,27 @@ def test_virtualfile_in_fail_non_valid_data(data):
                 z=data[:, 2],
                 data=data,
             )
+
```

Comment:
This test is added to address https://github.com/GenericMappingTools/pygmt/pull/3351#discussion_r1798573532.

This new test passes a string dtype numpy array into GMT C API, which contains longitude/latitude strings. The data kind is `"matrix"`, but since the data dtype is not in `iuf`, PyGMT will use `virtualfile_from_vectors` rather than `virtualfile_from_matrix` to pass the data into GMT C API. Ideally, we should check that the `virtualfile_from_vectors` is called once and `virtualfile_from_matrix` is not called, but I find it technically complicated with `unittest.mock`, so I'll leave it untested.

The function is marked as xfail with GMT 6.5 due to a newly found upstream bug at https://github.com/GenericMappingTools/gmt/pull/8600.

Reply:
Good catch, and thanks for adding the test!